core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -326,4 +322,8 @@ public Version indexVersionCreated() {
     public QueryParseContext parseContext() {
         return this.parseContext;
     }
+
+    public Client client() {
```

Comment:
wouldn't that method need a client to perform the GET request which isn't necessarily local to the shard?